iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -291,24 +299,51 @@ private async Task ReceiveMessagesAsync(CancellationToken cancellationToken)
 
         private async Task ReceiveMessageAndCompleteAsync()
         {
-            using Message receivedMessage = await s_deviceClient.ReceiveAsync(s_sleepDuration);
+            using var cts = new CancellationTokenSource(s_sleepDuration);
+            Message receivedMessage = null;
+            try
+            {
+                // AMQP library does not take a cancellation token but does take a time span
+                // so we'll call this API differently.
+                if (_amqpTransports.Contains(_transportType))
+                {
+                    receivedMessage = await s_deviceClient.ReceiveAsync(s_sleepDuration);
+                }
+                else
+                {
+                    receivedMessage = await s_deviceClient.ReceiveAsync(cts.Token);
+                }
+            }
+            catch (IotHubCommunicationException ex) when (ex.InnerException is OperationCanceledException)
+            {
+                // With MQTT, using a cancellation token, if it expires then the API throws this exception
```

Comment:
Should we add a output log here that the operation was cancelled likely due to no message being there to read or that the operation is taking longer than the set timeout?

Reply:
Sure